docs/angular/your-first-app/7-live-reload.md

```diff
@@ -23,92 +24,147 @@ We can also use it when developing on iOS and Android devices. This is particula
 Let’s use Live Reload to implement photo deletion, the missing piece of our Photo Gallery feature. Select your platform of choice (iOS or Android) and connect a device to your computer. Next, run either command in a terminal, based on your chosen platform:
 
 ```shell
-$ ionic cap run ios -l --external
+ionic cap run ios -l --external
 
-$ ionic cap run android -l --external
+ionic cap run android -l --external
 ```
 
 The Live Reload server will start up, and the native IDE of choice will open if not opened already. Within the IDE, click the Play button to launch the app onto your device.
 
 ## Deleting Photos
 
-With Live Reload running and the app open on your device, let’s implement photo deletion functionality. In your code editor (not Android Studio or Xcode), open `tab2.page.html` and add a new click handler to each `<ion-img>` element. When the app user taps on a photo in our gallery, we’ll display an [Action Sheet](https://ionicframework.com/docs/api/action-sheet) dialog with the option to either delete the selected photo or cancel (close) the dialog.
+With Live Reload running and the app open on your device, let’s implement photo deletion functionality.
+
+In `photo.service.ts`, add the `deletePhoto()` method. The selected photo is removed from the `photos` array first. Then, we use the Capacitor Preferences API to update the cached version of the `photos` array. Finally, we delete the actual photo file itself using the Filesystem API.
+
+```ts
+import { Injectable } from '@angular/core';
+import { Camera, CameraResultType, CameraSource, Photo } from '@capacitor/camera';
+import { Filesystem, Directory } from '@capacitor/filesystem';
+import { Preferences } from '@capacitor/preferences';
+import { Platform } from '@ionic/angular';
+import { Capacitor } from '@capacitor/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PhotoService {
+  // ...existing code...
+
+  // CHANGE: Add `deletePhoto()` method
+  public async deletePhoto(photo: UserPhoto, position: number) {
+    // Remove this photo from the Photos reference data array
+    this.photos.splice(position, 1);
+
+    // Update photos array cache by overwriting the existing photo array
+    Preferences.set({
+      key: this.PHOTO_STORAGE,
+      value: JSON.stringify(this.photos),
+    });
+
+    // Delete photo file from filesystem
+    const filename = photo.filepath.substr(photo.filepath.lastIndexOf('/') + 1);
```

Comment:
```suggestion
    const filename = photo.filepath.slice(photo.filepath.lastIndexOf('/') + 1);
```

`substr` is deprecated - optional change since it is in the [final code](https://github.com/ionic-team/tutorial-photo-gallery-angular/blob/fc17b81eafe00155c45803a5a1aca839e67ba148/src/app/services/photo.service.ts#L130)

Reply:
https://github.com/ionic-team/ionic-docs/pull/4157/commits/86f805e4e76dfbf5c34b552897daabcd0cf78b45